demos/minimal/actors/sample_processor.py

```diff
@@ -1,63 +1,77 @@
 from improv.actor import Actor
-import numpy as np
+from queue import Empty
 import logging
+import zmq
+import numpy as np
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
 class Processor(Actor):
-    """Sample processor used to calculate the average of an array of integers.
-
-    Intended for use with sample_generator.py.
+    """
+    Process data by scaling y coordinates by 2 and send it through zmq to be visualized.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def setup(self):
-        """Initializes all class variables.
-
-        self.name (string): name of the actor.
-        self.frame (ObjectID): StoreInterface object id referencing data from the store.
-        self.avg_list (list): list that contains averages of individual vectors.
-        self.frame_num (int): index of current frame.
         """
-
+        Creates and binds the socket for zmq and initializes processed data storage.
+        """
         self.name = "Processor"
-        self.frame = None
-        self.avg_list = []
-        self.frame_num = 1
+        self.processed_data = None  # Initialize variable to store processed data
+        self.frame = None  # Initialize variable to store the current frame
+
+        context = zmq.Context()
+        self.socket = context.socket(zmq.PUB)
+        self.socket.bind("tcp://127.0.0.1:5555")
+
         logger.info("Completed setup for Processor")
 
     def stop(self):
```

Comment:
This function should still have a doc string: 

`"""Trivial stop function for testing purposes."""`